api/v1alpha4/openstackcluster_webhook_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestOpenStackCluster_ValidateUpdate(t *testing.T) {
+	g := NewWithT(t)
+
+	tests := []struct {
+		name        string
+		oldTemplate *OpenStackCluster
+		newTemplate *OpenStackCluster
+		wantErr     bool
+	}{
+		{
+			name: "OpenStackCluster.Spec.IdentityRef with immutable spec",
+			oldTemplate: &OpenStackCluster{
+				Spec: OpenStackClusterSpec{
+					IdentityRef: &OpenStackIdentityReference{
+						Kind: "Secret",
+					},
+					CloudName: "foobar",
+				},
+			},
+			newTemplate: &OpenStackCluster{
+				Spec: OpenStackClusterSpec{
+					CloudName: "foobar",
+					IdentityRef: &OpenStackIdentityReference{
+						Kind: "foobar",
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "OpenStackCluster.Spec.IdentityRef with mutable spec",
+			oldTemplate: &OpenStackCluster{
+				Spec: OpenStackClusterSpec{
+					IdentityRef: &OpenStackIdentityReference{
+						Kind: "Secret",
+					},
+					CloudName: "foobar",
+				},
+			},
+			newTemplate: &OpenStackCluster{
+				Spec: OpenStackClusterSpec{
+					CloudName: "foobar",
+					IdentityRef: &OpenStackIdentityReference{
+						Kind: "Secret",
+					},
+				},
+			},
```

Comment:
These are the same. Did you intend to change IdentityRef.Name?

Reply:
You're totally right. This is a leftover from a previous test, but I will add the `IdentityRef.Name` 👍🏻